loadtest/control/simulcontroller/controller.go

```diff
@@ -48,16 +49,123 @@ func (c *SimulController) Run() {
 
 	defer c.sendStopStatus()
 
+	if resp := control.SignUp(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	if resp := control.Login(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+		c.connect()
+	}
+
+	if resp := c.reload(false); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	if resp := control.JoinTeam(c.user); resp.Err != nil {
+		c.status <- c.newErrorStatus(resp.Err)
+	} else {
+		c.status <- c.newInfoStatus(resp.Info)
+	}
+
+	actions := []userAction{
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.reload(false)
+			},
+			frequency: 1,
+		},
+		{
+			run:       control.JoinTeam,
+			frequency: 1,
+		},
+		{
+			run:       control.JoinChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchPosts,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchChannels,
+			frequency: 1,
+		},
+		{
+			run:       control.SearchUsers,
+			frequency: 1,
+		},
+		{
+			run:       control.ViewUser,
+			frequency: 1,
+		},
+		{
+			run:       control.CreatePost,
+			frequency: 1,
+		},
+		{
+			run:       control.AddReaction,
+			frequency: 1,
+		},
+		{
+			run:       control.UpdateProfileImage,
+			frequency: 1,
+		},
+		{
+			run:       control.CreateGroupChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.CreateDirectChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.ViewChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.LeaveChannel,
+			frequency: 1,
+		},
+		{
+			run:       control.RemoveReaction,
+			frequency: 1,
+		},
+	}
+
 	for {
-		// TODO: implement picking an action and run it.
+		action := pickAction(actions)
+
+		if resp := action.run(c.user); resp.Err != nil {
+			c.status <- c.newErrorStatus(resp.Err)
+		} else {
+			c.status <- c.newInfoStatus(resp.Info)
+		}
+
+		// TODO: make the following values configurable.
+		// Minimum idle time value in milliseconds.
+		minIdleMs := 1000
+		// Average idle time value in milliseconds.
+		avgIdleMs := 10000
```

Comment:
We can move this initializations out of the loop.

Reply:
Yes. We'll soon have to move these into a config file so I decided to keep everything close for now.

Reply:
Since it has a `TODO`, its OK 👍